File: dev-docs/snippets/database-type-configs/update.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Update an existing database type configuration. Pass `null` for any field
// to clear it; omit a field to leave it unchanged.
export async function update(
  client: JsBaoClient,
  databaseType: string,
  ruleSetId: string,
) {
  // #region example
  const config = await client.databaseTypeConfigs.update(databaseType, {
    ruleSetId,
    triggers: null,
    metadataAccess: "user.role == 'admin'",
  });
  // #endregion example
  return config;
}

File: dev-docs/snippets/database-type-configs/update.swift
import Foundation
import JsBaoClient

// Update an existing database type configuration. To clear a field, pass
// `NSNull()` (the untyped dict equivalent of JS `null`); omit a key to leave
// it unchanged.
func update(
  client: JsBaoClient,
  databaseType: String,
  ruleSetId: String
) async throws {
  // #region example
  let config = try await client.databaseTypeConfigs.update(
    databaseType: databaseType,
    params: [
      "ruleSetId": ruleSetId,
      "triggers": NSNull(),
      "metadataAccess": "user.role == 'admin'",
    ]
  )
  // #endregion example
  _ = config
}
